OpenRA.Mods.Common/Widgets/Logic/Editor/ActorEditLogic.cs

```diff
@@ -125,12 +125,15 @@ public ActorEditLogic(Widget widget, World world, WorldRenderer worldRenderer, D
 					return;
 				}
 
-				// Check for duplicate actor ID
-				if (CurrentActor.ID.Equals(actorId, StringComparison.OrdinalIgnoreCase))
+				// Check for duplicate ActorID
+				foreach (var kv in world.Map.ActorDefinitions)
 				{
-					if (editorActorLayer[actorId] != null)
+					if (kv.Key.ToString().ToLowerInvariant() == actorId.ToLowerInvariant() &&
+						CurrentActor.ID.ToString().ToLowerInvariant() != actorId.ToString().ToLowerInvariant())
```

Comment:
ID is already a string, so I am not sure what those `ToString()` are for.

Reply:
Corrected and pushed.